docs/_posts/2020-11-11-hudi-indexing-mechanisms.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "Apache Hudi Indexing mechanisms"
+excerpt: "Detailing different indexing mechanisms in Hudi and when to use each of them"
+author: sivabalan
+category: blog
+---
+
+
+## Introduction
+Hudi employs index to find and update the location of incoming records during write operations. To be specific, index assist in differentiating 
+inserts vs updates. This blog talks about different indices and when to each of them.
+
+Hudi dataset can be of two types in general, partitioned and non-partitioned. So, most index has two implementations, one for partitioned dataset 
+and another for non-partitioned called as global index.
+
+These are the types of index supported by Hudi as of now.
+
+- InMemory
```

Comment:
Even though the blog talks about only 3 of these, just to be comprehensive, have included InMemory also here.

Reply:
lets please remove InMemory as an option.. it's just something that is used by tests atm .